### PR TITLE
[ENG-15812] Adding lakeview sync tool module

### PR DESCRIPTION
**Pull request by pkgajulapalli**

### CODE REVIEW

**Comment by nimahajan** [comment]
Task linked: [ENG-15812 Add lakeview sync tool in lakeview repo](https://app.clickup.com/t/18029943/ENG-15812)

**Review by pkgajulapalli — COMMENTED** [review]
/push-image lakeview-sync-tool-test-0

**Review by pkgajulapalli — COMMENTED** [review]
/push-image lakeview-sync-tool-test-1

File: lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncConfigHolder.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+import com.beust.jcommander.ParametersDelegate;
+import org.apache.hudi.common.config.ConfigProperty;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.sync.common.HoodieSyncConfig;
+
+import java.util.List;
+
+import static org.apache.hudi.common.config.HoodieCommonConfig.BASE_PATH;
+
+public class LakeviewSyncConfigHolder {
```
Comment (sampan-s-nayak) [inline lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncConfigHolder.java:14]: nit (feel free to ignore): LakeviewSyncConfigStore instead?

File: lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncConfigHolder.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      .key("hoodie.meta.sync.lakeview.s3.region")
+      .noDefaultValue()
+      .markAdvanced()
+      .withDocumentation("S3 region associated with the table base path");
+
+  public static final ConfigProperty<String> LAKEVIEW_S3_ACCESS_KEY = ConfigProperty
+      .key("hoodie.meta.sync.lakeview.s3.accessKey")
+      .noDefaultValue()
+      .markAdvanced()
+      .withDocumentation("Access key required to access table base paths present in S3");
```
Comment (sampan-s-nayak) [inline lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncConfigHolder.java:63]: lets also call out that these are optional?

File: lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/utilities/IdentitySplitter.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package ai.onehouse.lakeview.sync.utilities;
+
+import com.beust.jcommander.converters.IParameterSplitter;
+
+import java.util.Collections;
+import java.util.List;
+
+public class IdentitySplitter implements IParameterSplitter {
```
Comment (sampan-s-nayak) [inline lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/utilities/IdentitySplitter.java:8]: is this used?

File: lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncTool.java [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+  @Override
+  public void syncHoodieTable() {
+    if (isLakeviewSyncToolEnabled && tableDiscoveryAndUploadJob != null) {
+      Future<?> future = executorService.submit(tableDiscoveryAndUploadJob::runOnce);
+      try {
+        if (timeoutInSeconds > 0) {
+          future.get(timeoutInSeconds, TimeUnit.SECONDS);
+        } else {
+          future.get();
```
Comment (sampan-s-nayak) [inline lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncTool.java:294]: should we consider having a default timeout? first time ingestion can take hours sometimes

File: lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/utilities/IdentitySplitter.java (hunk 3 above)
Comment (pkgajulapalli) [reply lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/utilities/IdentitySplitter.java:8]: yes

File: lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncConfigHolder.java (hunk 1 above)
Comment (pkgajulapalli) [reply lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncConfigHolder.java:14]: this is as per the OSS convention

File: lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncTool.java (hunk 4 above)
Comment (pkgajulapalli) [reply lakeview-sync-tool/src/main/java/ai/onehouse/lakeview/sync/LakeviewSyncTool.java:294]: we provide a timeout to be configured by users if needed. i'll check with Andy on the default timeout of this separately.

**Review by sampan-s-nayak — APPROVED** [review]

**Pull request merged**